src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -3492,27 +3499,64 @@ bool AdaptDispatch::SetColorTableEntry(const size_t tableIndex, const DWORD dwCo
 }
 
 // Method Description:
-// - Sets the default foreground color to a new value
-// Arguments:
-// - dwColor: The new RGB color value to use, as a COLORREF, format 0x00BBGGRR.
+// - Sets one Xterm Color Resource such as Default Foreground, Background, Cursor
 // Return Value:
 // True if handled successfully. False otherwise.
-bool AdaptDispatch::SetDefaultForeground(const DWORD dwColor)
+bool AdaptDispatch::SetXtermColorResource(const DispatchTypes::XtermColorResource resource, const DWORD color)
 {
-    _renderSettings.SetColorAliasIndex(ColorAlias::DefaultForeground, TextColor::DEFAULT_FOREGROUND);
-    return SetColorTableEntry(TextColor::DEFAULT_FOREGROUND, dwColor);
+    assert(static_cast<size_t>(resource) >= 10);
```

Comment:
This protects the developer. `{Set,Request}XtermColorResource` are amply guarded by the `switch` in `OutputStateMachineEngine`.